network-store-server/src/main/java/com/powsybl/network/store/server/LimitsHandler.java

```diff
@@ -627,27 +627,43 @@ private <T extends IdentifiableAttributes> void insertTombstonedPermanentLimits(
         }
     }
 
-    public Map<OperationalLimitsGroupIdentifier, OperationalLimitsGroupAttributes> convertLimitInfosToOperationalLimitsGroupMap(String equipmentId, LimitsInfos limitsInfos) {
-        Map<OperationalLimitsGroupIdentifier, OperationalLimitsGroupAttributes> operationalLimitGroups = new HashMap<>();
+    private boolean isMapContainsOperationalLimitsGroup(Map<String, Map<Integer, Map<String, OperationalLimitsGroupAttributes>>> map, String branchId, Integer side, String groupId) {
+        return map.containsKey(branchId) && map.get(branchId).containsKey(side) && map.get(branchId).get(side).containsKey(groupId);
+    }
+
+    private void addElementToOperationalLimitsGroupMap(Map<String, Map<Integer, Map<String, OperationalLimitsGroupAttributes>>> map, String branchId, Integer side, String groupId, OperationalLimitsGroupAttributes operationalLimitsGroupAttributes) {
+        map.computeIfAbsent(branchId, k -> new HashMap<>())
+            .computeIfAbsent(side, k -> new HashMap<>())
+            .put(groupId, operationalLimitsGroupAttributes);
+    }
+
+    private OperationalLimitsGroupAttributes getElementFromOperationalLimitsGroupMap(Map<String, Map<Integer, Map<String, OperationalLimitsGroupAttributes>>> map, String branchId, Integer side, String groupId) {
+        if (isMapContainsOperationalLimitsGroup(map, branchId, side, groupId)) {
+            return map.get(branchId).get(side).get(groupId);
+        }
+        return null;
+    }
```

Comment:
```suggestion
    private Optional<OperationalLimitsGroupAttributes> getElementFromOperationalLimitsGroupMap(Map<String, Map<Integer, Map<String, OperationalLimitsGroupAttributes>>> operationalLimitsGroupMap, String branchId, Integer side, String groupId) {
        return Optional.ofNullable(operationalLimitsGroupMap.get(branchId))
            .map(sideMap -> sideMap.get(side))
            .map(groupMap -> groupMap.get(groupId));
    }
```
as it's used as an optional later ?